### PR TITLE
add preinstall guide

### PR DESCRIPTION
**Pull request by Redecorating**
includes steps for disabling secure boot, creating an installation usb, etc. This will allow reduction of the amount of duplicated instructions in each distro's install guide.

I haven't edited the other distro guides to rely on these new consolidated instructions so that needs doing (if maintainers want to do that they should be able to add commits to this branch)

### CODE REVIEW

**Comment by Redecorating** [comment]
also @kekrby for the nixos iso's here https://github.com/t2linux/nixos-t2-iso/releases could the release description link to the section in the readme about combining the iso parts? or maybe have those instructions added to the release description. Otherwise the link I put on this wiki page could be changed to the readme instead of the releases page

File: docs/guides/preinstall.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+| ------------------ | --------------------------- |
+| Arch Linux         | https://github.com/t2linux/archiso-t2/releases/latest |
+| EndeavourOS        | https://github.com/t2linux/EndeavourOS-ISO-t2/releases/latest |
+| Fedora Linux       | https://github.com/mikeeq/mbp-fedora |
+| Gentoo             | Please refer to this [page](https://wiki.t2linux.org/distributions/gentoo/installation/) |
+| Manjaro            | https://github.com/NoaHimesaka1873/manjaroiso-t2/releases/latest |
+| NixOS              | https://github.com/kekrby/nixos-t2-iso/releases/latest TODO: Instructions for combining iso parts |
+| Ubuntu             | https://github.com/AdityaGarg8/T2-Ubuntu/releases/latest |
+
+For other distros, you can download the distro's normal install ISO, but you will have to use an external USB keyboard and mouse for this install process, and you may need a wired internet connection.
```
Comment (AdityaGarg8) [inline docs/guides/preinstall.md:37]: Probably add troubleshooting if grub install failed, with efi=noruntime as a kernel parameter

**Review by AdityaGarg8 — CHANGES_REQUESTED** [review]

File: docs/guides/preinstall.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+!!! hint "`dd`"
+    If you are familiar with the `dd` tool and the command line, you can use that instead. Use `diskutil list` to find the correct disk, then `sudo dd if=path/to/filename.iso of=/dev/rdiskX bs=1m`. Press `control-T` while it is running to make it show its current progress. **Make sure you select the correct disks.**
+
+While the installation image is being written to the USB, you can copy Wi-Fi firmware, but don't follow the steps after that until it has finished writing the image.
+
+## Copy Wi-Fi firmware
+
+Linux's Wi-Fi driver uses the same Wi-Fi firmware files as macOS, so we copy these files from macOS to the EFI partition where Linux can access these files and then install them.
+
+Follow the instructions in this section for on macOS [here](https://wiki.t2linux.org/guides/wifi/#on-macos). Don't boot into Linux and follow the subsequent instructions yet.
```
Comment (AdityaGarg8) [inline docs/guides/preinstall.md:52]: Don't boot into Linux ...

This line is a bit confusing. What exactly we wish to convey here?

Comment (Redecorating) [reply docs/guides/preinstall.md:52]: at this point we just want them to put the firmware on the efi partition, so the instructions in that section of the wifi guide

4. When the script shall run successfully, it shall ask you to follow either of the two options mentioned in the On Linux section, on Linux.
5. Boot into Linux.

Shouldn't be followed yet, thus don't boot linux yet

Comment (AdityaGarg8) [reply docs/guides/preinstall.md:52]: Well obviously we cannot boot into Linux if it’s not installed. Maybe we can write that the Linux bit can be followed after Linux has installed successfully.

File: docs/guides/preinstall.md (hunk 1 above)
Comment (Redecorating) [reply docs/guides/preinstall.md]: d03aadef42316831995bd1b0dfc562ca233ae497

> Additionally, if you later have issues with installing the bootloader, try the `efi=noruntime` kernel parameter.

File: docs/guides/preinstall.md (hunk 2 above)
Comment (Redecorating) [reply docs/guides/preinstall.md]: d03aadef42316831995bd1b0dfc562ca233ae497

> The sections of that page for on Linux can be followed later, once you've successfully installed Linux.

Comment (AdityaGarg8) [reply docs/guides/preinstall.md]: Instead of `for on`, could we do `to be followed on`?

File: docs/guides/preinstall.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Pre install steps
+
+These steps are common to most distros. Please refer to both this page and this wiki's install page for your distro.
+
+## Make disk space with Bootcamp Assistant
```
Comment (AdityaGarg8) [inline docs/guides/preinstall.md:5]: Has anyone tried it? Iirc, Bootcamp partitions only when you provide it with the Windows ISO.

**Review by AdityaGarg8 — APPROVED** [review]

**Comment by AdityaGarg8** [comment]
I did some changes, maybe you could review them as well

**Comment by Redecorating** [comment]
> I did some changes, maybe you could review them as well

That all looks good to me!

File: docs/guides/preinstall.md (hunk 3 above)
Comment (AdityaGarg8) [reply docs/guides/preinstall.md:5]: I had recently reinstalled windows and it asked for the ISO first.

Pretty much similar to this https://youtu.be/Hmm9Q-T0oTo

**Comment by Redecorating** [comment]
@NoaHimesaka1873 could you check 12bf95fe89f4d561ff8bd344bb9a11e7b374dcfc when you have time?

**Comment by NoaHimesaka1873** [comment]
@Redecorating Seems fine.

**Comment by EmreTech** [comment]
> The Gentoo guide needs a lot of work to be done. @EmreTech , or someone else should get it done imo.
> 
> 

I'm not actively maintaining the Gentoo guide or T2 Gentoo repos atm, so I would be glad if someone else fixed the guide!

File: docs/guides/postinstall.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+Many distro maintainers provide compiled kernels which can be installed on your Linux installation. Following are the links to the repos providing such kernels :-
+
+| Linux Distribution   | Kernel with T2 support |
+| -------------------- | ---------------------- |
+| Arch based distros   | https://github.com/Redecorating/linux-t2-arch |
+| Fedora               | https://github.com/mikeeq/mbp-fedora-kernel |
+| Gentoo               | https://github.com/t2linux/T2-Gentoo-Kernel |
+| Manjaro              | https://github.com/NoaHimesaka1873/manjaro-kernel-t2 |
+| NixOS                | TODO |
```
Comment (AdityaGarg8) [inline docs/guides/postinstall.md:27]: @kekrby is there any separate repo for nixOS kernels?

@NoaHimesaka1873 I guess you maintain xanmod kernels as well. Could you tell me where should I place them in this table.

File: docs/guides/preinstall.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+### Copying the ISO to the USB
+
+You can use [Belana Etcher](https://www.balena.io/etcher/) to copy the Linux install image to your USB.
+
+!!! hint "`dd`"
+    If you are familiar with the `dd` tool and the command line, you can use that instead. Use `diskutil list` to find the correct disk, then `sudo dd if=path/to/filename.iso of=/dev/rdiskX bs=1m`. Press `control-T` while it is running to make it show its current progress. **Make sure you select the correct disks.**
+
+While the installation image is being written to the USB, you can copy Wi-Fi firmware, but don't follow the steps after that until it has finished writing the image.
+
+## Copy Wi-Fi firmware
```
Comment (NoaHimesaka1873) [inline docs/guides/preinstall.md:48]: You should mention that you skip this for Arch, EOS, and Manjaro.

**Review by NoaHimesaka1873 — CHANGES_REQUESTED** [review]

**Review by NoaHimesaka1873 — APPROVED** [review]
Seems good!

File: docs/guides/postinstall.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+Many distro maintainers provide compiled kernels which can be installed on your Linux installation. Following are the links to the repos providing such kernels :-
+
+| Linux Distribution                  | Kernel with T2 support |
+| ----------------------------------- | ---------------------- |
+| Arch based distros                  | https://github.com/Redecorating/linux-t2-arch |
+| Arch based distros (Xanmod kernels) | https://github.com/NoaHimesaka1873/linux-xanmod-edge-t2 |
+| Fedora                              | https://github.com/mikeeq/mbp-fedora-kernel |
+| Gentoo                              | https://github.com/t2linux/T2-Gentoo-Kernel |
+| Manjaro                             | https://github.com/NoaHimesaka1873/manjaro-kernel-t2 |
+| NixOS                               | TODO |
```
Comment (kekrby) [inline docs/guides/postinstall.md:28]:
```suggestion
| NixOS                               | https://github.com/kekrby/nixos-hardware |
```

**Comment by AdityaGarg8** [comment]
@Redecorating, except for the bootcamp resize partition thing, rest everything done.

File: docs/guides/preinstall.md (hunk 3 above)
Comment (Redecorating) [reply docs/guides/preinstall.md:5]: hmm i just tried it and it seems like either they changed it in an update, or i was remembering wrong.

I think that means we can take out this whole section then, which makes things simpler.

Comment (AdityaGarg8) [reply docs/guides/preinstall.md:5]: You may do the editing then.

Comment (AdityaGarg8) [reply docs/guides/preinstall.md:5]: Also review the rest of the PR if you get time.

**Review by Redecorating — COMMENTED** [review]
it all looks good to me currently so we should be able to merge?

**Comment by Redecorating** [comment]
and thank you everyone for helping with this one :)

**Comment by AdityaGarg8** [comment]
Let’s merge it!

**Comment by AdityaGarg8** [comment]
Squash the commits or not?
I leave it to you @Redecorating

**Pull request merged**